### PR TITLE
fix(stdlib): fix issues when using LVGL TLSF memory pool to manage more than 4 GiB of memory

### PR DESCRIPTION
**Pull request by MouriNaruto**
### Description of the feature or fix

People will know that I'm developing UEFI applications with LVGL in recent days after read https://github.com/lvgl/lvgl/issues/5391#issuecomment-1909771955.

Because UEFI memory pool will cause many memory fragments. So, my project uses LVGL builtin TLSF memory pool to manage the memory. And I found if I use LVGL TLSF memory pool to manage more than 4 GiB of memory, I cannot get the correct result via  lv_mem_monitor API.

I also found the original LVGL implementation have considered about the 64-bit environments and many lv_mem APIs use size_t to manage size. So, I think it should not break the ABI for 32-bit environments if we use size_t type instead of uint32_t type to fix the issue. (For 64-bit LVGL users needs to recompile. But I think most people don't use 64-bit LVGL instance in the current stage. It should be acceptable.)

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.10](https://github.com/szepeviktor/astyle/releases/tag/v3.4.10) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

Kenji Mouri

### CODE REVIEW

**Comment by MouriNaruto** [comment]
I think there are some issues should be also fixed for LVGL. (Need some time to do.)

Kenji Mouri

**Comment by MouriNaruto** [comment]
It looks like all CIs are passed. It should be continued.

Kenji Mouri

**Review by XuNeo — DISMISSED** [review]

File: src/stdlib/lv_mem.c [hunk 1]
```diff
@@ -74,9 +74,9 @@ void * lv_malloc(size_t size)
 #if LV_LOG_LEVEL <= LV_LOG_LEVEL_INFO
         lv_mem_monitor_t mon;
         lv_mem_monitor(&mon);
-        LV_LOG_INFO("used: %6d (%3d %%), frag: %3d %%, biggest free: %6d",
-                    (int)(mon.total_size - mon.free_size), mon.used_pct, mon.frag_pct,
-                    (int)mon.free_biggest_size);
+        LV_LOG_INFO("used: %zu (%3d %%), frag: %3d %%, biggest free: %zu",
+                    (size_t)(mon.total_size - mon.free_size), mon.used_pct, mon.frag_pct,
```
Comment (FASTSHIFT) [inline src/stdlib/lv_mem.c]: remove size_t cast.

File: src/stdlib/lv_mem.c [hunk 2]
```diff
@@ -103,9 +103,9 @@ void * lv_malloc_zeroed(size_t size)
...
```
Comment (FASTSHIFT) [inline src/stdlib/lv_mem.c]: remove size_t cast.

**Comment by MouriNaruto** [comment]
image
Here is the screenshot of just using LVGL TLSF memory pool to manage 6 GiB of memory in my test virtual machine with this fix.

Kenji Mouri

**Review by FASTSHIFT — CHANGES_REQUESTED** [review]

File: src/stdlib/lv_mem.c (hunk 1 above)
Comment (MouriNaruto) [reply src/stdlib/lv_mem.c]: OK. I have modified it.

Kenji Mouri

File: src/stdlib/lv_mem.c (hunk 2 above)
Comment (MouriNaruto) [reply src/stdlib/lv_mem.c]: OK. I have modified it.

Kenji Mouri

**Review by FASTSHIFT — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]
Awesome, thank you!

**Pull request merged**